rsocket-routing-broker/src/main/java/io/rsocket/routing/broker/locator/WeightedStatsAwareRSocket.java

```diff
@@ -0,0 +1,19 @@
+package io.rsocket.routing.broker.locator;
+
+import io.rsocket.RSocket;
+import io.rsocket.loadbalance.WeightedStats;
+import io.rsocket.util.RSocketProxy;
+
+public class WeightedStatsAwareRSocket extends RSocketProxy {
+
+	private final WeightedStats weightedStats;
+
+	public WeightedStatsAwareRSocket(RSocket source, WeightedStats weightedStats) {
+		super(source);
+		this.weightedStats = weightedStats;
+	}
+
+	public WeightedStats weightedStats() {
+		return weightedStats;
+	}
+}
```

Comment:
I think this class does not need to exist as public at all. Instead it can implement the new `WeightedStats` interface and  so code looking for stats can check `rsocket instanceof WeightedStats`. 

It can be applied from `BrokerAutoConfiguration` where it can be nested or package private and likewise it can be applied from `ClusterJoinListener`. To avoid duplication perhaps we could expose a method on `WeightedStatsIntereceptor` which helps to create it and keep it internal and encapsulated.